### PR TITLE
Disable Hideout Functionality

### LINKED ISSUE #582: Disable All Data Updates in the Hideout class
# For New Contributors
[How to contribute](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/wiki/Getting-Started-as-a-Contributor)

## Description

Disable all functionality that creates, updates, or deletes any Hideout or data within the Hideout class using [Harmony patches](https://harmony.pardeike.net/articles/patching.html).

Essentially disable all functionality without game crashing changes.

1. Skip all setters
2. Disable any tick functions
3. Any method that modifies data in the class should be skipped or patched

## Intended Design

N/A

## Location

Create a new branch based from `development`

## Related Issues

N/A

## Requirements

N/A

## Additional information

N/A

## Definition of Done
- [ ] Class level comments exist for all new classes.
- [ ] XUnit tests exist for every method that does not require the game to be ran.

### PR DESCRIPTION
**Pull request by garrettluskey**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] All new classes have class-level documentation comments, if there are any at all
- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:

## What is the current behaviour?

Currently Hideouts are enabled which makes it harder to develop other features

## What is the new behaviour?

Resolves #582

Disables all Hideout functionality

## Other information

### CODE REVIEW

File: source/GameInterface/Services/GameDebug/Patches/PartyVisibilityPatch.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    /// Patches for party visibility
+    /// </summary>
+    [HarmonyPatch(typeof(MobileParty))]
+    internal class PartyVisibilityPatch
+    {
+        public static bool AllPartiesVisible = false;
+
+        private static Action<MobileParty, bool> _isVisibleSetter = typeof(MobileParty)
+            .GetField("_isVisible", BindingFlags.NonPublic | BindingFlags.Instance)
+            .BuildUntypedSetter<MobileParty, bool>();
```
Comment (orfolei) [inline source/GameInterface/Services/GameDebug/Patches/PartyVisibilityPatch.cs:21]: Have you considered using this Harmony feature instead?
https://harmony.pardeike.net/articles/patching-injections.html#___fields

File: source/GameInterface/Services/GameDebug/Interfaces/GameDebugInterface.cs [hunk 2]
```diff
@@ -30,10 +35,10 @@ public void LoadDebugGame()
             GameLoopRunner.RunOnMainThread(InternalLoadDebugGame, false);
         }
 
-        private static readonly PropertyInfo FileDriver_SavePath = typeof(FileDriver).GetProperty("SavePath", BindingFlags.NonPublic | BindingFlags.Static);
-        private static readonly PlatformDirectoryPath SaveDir = (PlatformDirectoryPath)FileDriver_SavePath?.GetValue(null);
-        private static readonly PlatformFilePath SavePath = new PlatformFilePath(SaveDir, $"{LOAD_GAME}.sav");
-        private static readonly string FullSavePath = TaleWorlds.Library.Common.PlatformFileHelper?.GetFileFullPath(SavePath);
+        private static PropertyInfo FileDriver_SavePath => typeof(FileDriver).GetProperty("SavePath", BindingFlags.NonPublic | BindingFlags.Static);
```
Comment (orfolei) [inline source/GameInterface/Services/GameDebug/Interfaces/GameDebugInterface.cs:38]: I assume the right-hand side is evaluated every time an expression-bodied property is accessed? Not that it matters much here given the context.

**Review by orfolei — APPROVED** [review]

File: source/GameInterface/Services/GameDebug/Patches/PartyVisibilityPatch.cs (hunk 1 above)
Comment (garrettluskey) [reply source/GameInterface/Services/GameDebug/Patches/PartyVisibilityPatch.cs:21]: This untyped setter is probably faster

File: source/GameInterface/Services/GameDebug/Interfaces/GameDebugInterface.cs (hunk 2 above)
Comment (garrettluskey) [reply source/GameInterface/Services/GameDebug/Interfaces/GameDebugInterface.cs:38]: The reason I changed this was to reduce memory overhead. May be negligible though

**Pull request merged**